ionoscloud/resource_server.go

```diff
@@ -1032,14 +1046,14 @@ func resourceServerUpdate(ctx context.Context, d *schema.ResourceData, meta inte
 				return diag.FromErr(err)
 			}
 
-			_, primaryNicOk := d.GetOk("primary_nic")
-			if createdNic.Id != nil && !updateNic && !primaryNicOk {
+			if createdNic.Id != nil && createNic {
```

Comment:
@cristiGuranIonos not sure here, but I assume that if `createNic` is true, then `createdNic` will be always different than `nil`, so not sure if the first check related to the ID makes sense.

Reply:
to make sure it doesn't crash and it throws a normal error.